tools/src/main/scala/scala/scalanative/build/Logger.scala

```diff
@@ -0,0 +1,61 @@
+package scala.scalanative.build
+
+import java.lang.System.{err, out, lineSeparator => nl}
+
+import scala.sys.process.ProcessLogger
+
+/** A `Logger` is in charge of collecting log messages. */
+trait Logger {
+
+  /** Logs `msg` at the debug level. */
+  def debug(msg: String): Unit
+
+  /** Logs `msg` at the info level. */
+  def info(msg: String): Unit
+
+  /** Logs `msg` at the warn level. */
+  def warn(msg: String): Unit
+
+  /** Logs `msg` at the errro level. */
+  def error(msg: String): Unit
+
+  /** Logs at the debug level that the command `cmd` will start running. */
+  def running(cmd: Seq[String]): Unit
+
+  /** Executes `f` and logs at the info level how much long it took. */
+  def time[T](msg: String)(f: => T): T
```

Comment:
`running` and `time` don't have to be abstract here. It would be cleaner to move implementation for those (in `new Logger { ... }`) into the trait itself.